tinyagentos/routes/decisions.py

```
@@ -292,6 +293,53 @@ async def _apply_execution_grant(request: Request, decision: dict, value) -> Non
+    if meta.get("kind") != "delegation_gate":
+        return
+    policies = getattr(request.app.state, "execution_policies", None)
+    from_agent = meta.get("from_agent")
+    to_agent = meta.get("to_agent")
+    if policies is None or not from_agent or not to_agent:
+        return
+    approved = value == "approve"
+    if approved:
+        try:
+            await policies.add_grant(from_agent, "delegate", decision.get("id"))
+        except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)